crypto/s2n_certificate.c

```diff
@@ -619,6 +619,9 @@ void *s2n_cert_chain_and_key_get_ctx(struct s2n_cert_chain_and_key *cert_and_key
 
 s2n_pkey_type s2n_cert_chain_and_key_get_pkey_type(struct s2n_cert_chain_and_key *chain_and_key)
 {
+    if (!S2N_OBJECT_PTR_IS_READABLE(chain_and_key->cert_chain->head)) {
+        return S2N_PKEY_TYPE_UNKNOWN;
+    }
```

Comment:
We generally don't actually use that macro, although there's no reason not to.
And if you're fixing this one, you might also consider checking chain_and_key and chain_and_key->cert_chain, since they could also lead to a null dereference.

```suggestion
    if (chain_and_key == NULL
         || chain_and_key->cert_chain == NULL
         || chain_and_key->cert_chain->head == NULL)) {
        return S2N_PKEY_TYPE_UNKNOWN;
    }
```

Reply:
Ok, changed. I would have preferred to use one of the `ENSURE_REF` macros, but since this function returns an enum, that triggered warnings.